alternator/serialization.cc

```diff
@@ -245,6 +245,25 @@ rjson::value deserialize_item(bytes_view bv) {
     return deserialized;
 }
 
+// If the value in bv (supposedly created earlier by serialize_item()) has the
+// right type "expected_type", this function returns the value as a serialized
+// Scylla type. If the type doesn't match, returns a null optional.
+// In the current implementation, this function is much more efficient than
+// deserialize_item() above because it avoids transformation to/from JSON.
+// However, serialized_value_if_type() is only implemented for the natively-
+// serialized types S, B and N.
+std::optional<bytes> serialized_value_if_type(bytes_view bv, alternator_type expected_type) {
```

Comment:
So this function behaves the same way the `std::get(std::variant)` behaves, right? If so, I think you could make your function look closer to this implementation, e.g. `get<expected_type>(bv)`. The motivation is that, personally, I don't find the current name much informative, secondly, it'd be more obvious what the function does if it's closer to something I'm familiar with already.
Either way, it's just an idea.